### PR TITLE
docs: refactor sdk compatibility

### PR DESCRIPTION
**Pull request by dabeeeenster**
- Also bump Docusaurus
- Also fix a bunch of links

### CODE REVIEW

File: docs/docs/basic-features/managing-segments.md [hunk 1]
```diff
@@ -240,3 +240,30 @@ More simply, the order of precedence is:
 1. Identity
 2. Segment
 3. Flag
+
+## SDK Compatibility
+
+### `In` Segment operator
```
Comment (matthewelwell) [inline docs/docs/basic-features/managing-segments.md]: Hmm... I actually meant more that we should create a section for the `IN` operator with the other operators, e.g. the Modulo operator has it's own section [here](https://github.com/Flagsmith/flagsmith/pull/3371/files#diff-5c8f900088e9b7550b4df2e8d2db0ad8fc3f9c1c3c99e295688043f7cad924ebR190). I think we should add a new section for the IN operator and define the compatibility there.

Comment (dabeeeenster) [reply docs/docs/basic-features/managing-segments.md]: OK Yep - have shifted some other stuff around its more logically laid out now.

File: docs/docs/basic-features/managing-segments.md [hunk 2]
```diff
@@ -77,35 +79,36 @@ individual value weightings as part of a Segment override.
 :::important
 
 Some of the operators in local evaluation mode are only supported in later SDK versions, see
-[SDK Compatibility](../clients/overview.md#sdk-compatibility).
+[SDK Compatibility](#sdk-compatibility).
```
Comment (matthewelwell) [inline docs/docs/basic-features/managing-segments.md]: I think this link is no longer relevant, right?

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: docs/docs/basic-features/managing-segments.md (hunk 2 above)
Comment (dabeeeenster) [reply docs/docs/basic-features/managing-segments.md]: yep pushed

File: docs/docs/basic-features/managing-segments.md [hunk 3]
```diff
@@ -118,7 +114,31 @@ contains the tenancy ID of that user.
...
-### SemVer-aware operators
+:::important
+
+Earlier SDK versions will not work in local evaluation mode if your environment has segments with the `In` operator
+defined.
+
+To keep local evaluation from breaking, please ensure you have your SDK versions updated before you add such segments to
+your environment.
+
+:::
```
Comment (matthewelwell) [inline docs/docs/basic-features/managing-segments.md:125]: It's kinda odd that we have this alert here but not in the Modulo tab. We should probably standardise this in some way.

**Review by matthewelwell — APPROVED** [review]
Approved because I feel like this review cycle is becoming tedious but I've added one comment that I think we should consider addressing.

File: docs/docs/basic-features/managing-segments.md (hunk 3 above)
Comment (dabeeeenster) [reply docs/docs/basic-features/managing-segments.md:125]: OK added to the modulo and merged

**Pull request merged**